scripts/develop-module.js

```diff
@@ -0,0 +1,299 @@
+const fs = require('fs');
+const path = require('path');
+const arg = require('arg');
+const log = require('./log');
+
+const DS_VERSION = 'DS6.5.v1.01';
+
+class DevelopModuleError extends Error {
+    constructor(message) {
+        super(message);
+    }
+}
+
+const args = arg({
+    '--module-name': String,
+    '--help': Boolean
+});
+const moduleName = args['--module-name'];
+const shouldHelp = args['--help'];
+
+function run() {
+    try {
+        validateArguments();
+        processRequest();
+    } catch (error) {
+        if (!(error instanceof DevelopModuleError)) {
+            throw error;
+        }
+
+        log.error(error.message);
+    }
+}
+
+function validateArguments() {
+    if (shouldHelp || moduleName) {
+        return;
+    }
+
+    throw new DevelopModuleError('Incorrect usage! Pass --help flag for help.');
+}
+
+function processRequest() {
+    if (shouldHelp) {
+        processHelpRequest();
+    } else if (moduleName) {
+        processNewModuleRequest();
+    } else {
+        throw new DevelopModuleError('Script Error! Unknown flag that passed validation to process request.');
+    }
+}
+
+function processHelpRequest() {
+    log.info(`
+USAGE:
+    --module-name
+                                The new module that you want to generate the boilerplate code for.
+                                Eg: node scripts/develop-module.js --module-name=toast-message
+    --help
+                                Used to generate this help message
+                                Eg: node scripts/develop-module.js --help
+    `);
+}
+
+function processNewModuleRequest() {
+    const newModuleData = {
+        id: moduleName.replace(/\s/g, '-').toLowerCase(),
+        moduleName
+    };
+
+    addStyles(newModuleData);
+    addDocs(newModuleData);
+    addTests(newModuleData);
+
+    log.info('Generated boilerplate code for the new module %s', moduleName);
+    log.info('Add documentation in file ~ %s', newModuleData.moduleDocFilePath);
+    log.info('Add styles in folder ~ %s', newModuleData.srcFolderPath);
+    log.info('Add tests in file ~ %s', newModuleData.testFilePath);
+}
+
+function addStyles(newModuleData) {
+    addSourceStyles(newModuleData);
+    addSourceBundleStyles(newModuleData);
+    addPackagingStyles(newModuleData);
+}
+
+function addDocs(newModuleData) {
+    addDocsModuleDoc(newModuleData);
+    addDocsModuleList(newModuleData);
+    addDocsMain(newModuleData);
+    addDocsIndex(newModuleData);
+}
+
+function addSourceStyles(newModuleData) {
+    const srcFolderPath = path.join(__dirname, '..', 'src', 'less', newModuleData.id);
```

Comment:
Can we add a base path or something? I see this same __dirname, '..' src less used in so many places over and over

What I did in the other script is something like this (in global, so I can use it anywhere else)
```
const currentDir = path.dirname(__dirname);
const svgDir = path.resolve(currentDir, 'src', 'svg');
```
Then you can join using that last variable. 
I would do this for all the common variables, and the ones you don't have to, then use `currentDir` or something

Reply:
Will do. I overlooked this as I refactored the code.